src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,594 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Security;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+using JSObject = Interop.JavaScript.JSObject;
+using JSException = Interop.JavaScript.JSException;
+using HostObject = Interop.JavaScript.HostObject;
+using Uint8Array = Interop.JavaScript.Uint8Array;
+using Function = Interop.JavaScript.Function;
+
+namespace System.Net.Http
+{
+
+    // **Note** on `Task.ConfigureAwait(continueOnCapturedContext: true)` for the WebAssembly Browser.
+    // The current implementation of WebAssembly for the Browser does not have a SynchronizationContext nor a Scheduler
+    // thus forcing the callbacks to run on the main browser thread.  When threading is eventually implemented using
+    // emscripten's threading model of remote worker threads, via SharedArrayBuffer, any API calls will have to be
+    // remoted back to the main thread.  Most APIs only work on the main browser thread.
+    // During discussions the concensus has been that it will not matter right now which value is used for ConfigureAwait
+    // we should put this in place now.
+    internal partial class BrowserHttpHandler : HttpMessageHandler
+    {
+        // This partial implementation contains members common to Browser WebAssembly running on .NET Core.
+        private static readonly JSObject? s_fetch = (JSObject)Interop.Runtime.GetGlobalObject("fetch");
+        private static readonly JSObject? s_window = (JSObject)Interop.Runtime.GetGlobalObject("window");
+
+        /// <summary>
+        /// Gets whether the current Browser supports streaming responses
+        /// </summary>
+        private static bool StreamingSupported { get; }
+
+        static BrowserHttpHandler()
+        {
+            using (var streamingSupported = new Function("return typeof Response !== 'undefined' && 'body' in Response.prototype && typeof ReadableStream === 'function'"))
+                StreamingSupported = (bool)streamingSupported.Call();
+        }
+
+        public bool UseCookies
+        {
+            get => throw new PlatformNotSupportedException("Property UseCookies is not supported.");
+            set => throw new PlatformNotSupportedException("Property UseCookies is not supported.");
```

Comment:
Do we need all these unique strings?  I don't have a strong opinion, but I'd think for size reasons we'd prefer to just do `throw new PlatformNotSupportException()` and rely on the stack trace to inform of the problematic property.  @marek-safar?

Reply:
Yes, my preference is for sure `throw PlatformNotSupportedException();`